src/hkdf.rs

```diff
@@ -15,21 +15,23 @@ use generic_array::{ArrayLength, GenericArray};
 use hmac::{Hmac, Mac};
 
 pub struct Hkdf<D>
-    where D: Digest + Default,
+    where D: Digest,
           D::OutputSize: ArrayLength<u8>
 {
     pub prk: GenericArray<u8, D::OutputSize>,
 }
 
 impl<D> Hkdf<D>
-    where D: Digest + Default,
+    where D: Digest,
           D::OutputSize: ArrayLength<u8>
 {
     pub fn new(ikm: &[u8], salt: &[u8]) -> Hkdf<D> {
-        let mut hmac = Hmac::<D>::new(salt);
+        // Hmac::new() is now defined to return a Result, but as far as I can
+        // tell, it can only return Ok.
```

Comment:
@warner could you update this comment. Maybe `Since hmac v0.5.0 'new' returns a Result` ? (I think this will change more in the next version of hmac lib)